File: 00-每日一题/20200220_11.py
# 对撞指针
# 固定一端， 高的往里面走，只会容积变小， 短在高不变，宽有减小

class Solution:
    def maxArea(self, height):
        length = len(height)
        left = 0
        right = length-1
        res = 0
        while left < right:
            min_side = min(height[left], height[right])
            res = max(res, min_side * (right-left))
            if min_side == height[left]:
                left += 1
            else:
                right -= 1

        return res

if __name__ == '__main__':

    obj = Solution()
    while True:
        nums_str = input().strip().split()
        nums = list(map(int, nums_str))
        res = obj.maxArea(nums)
        print(res)

File: 00-每日一题/20200220_11.cpp
//
// Created by zanbo on 2020/2/20.
//

class Solution {
public:
    int maxArea(vector<int>& height) {
        int i=0;
        int j=height.size()-1;
        int res = min(height[i],height[j])*(j-i);
        while(i<j)
        {
            res = max(res,min(height[i],height[j])*(j-i));
            if(height[i]<=height[j])
                i++;
            else
                j--;
        }
        return res;
    }
};

